File: cpp/a13.cpp
#include <iostream>

using namespace std;

long n, k;
long a[100009];

int main() {

    cin >> n >> k;
    for (int i = 0; i < n; i++)
    {
        cin >> a[i];
    }

    long long sum = 0;
    int j = 1;
    for (int i = 0; i < n-1; i++)
    {
        while((j < n) && (a[j] <= a[i] + k)) {
            j++;
        }
        sum += j - i - 1;
    }
    
    cout << sum;
    return 0;
}

File: rust/a13.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        k: u64,
        a: [u64;n],
    }

    let mut sum = 0;
    let mut j: usize = 0;
    for i in 0..n-1 {
        while j < n && a[j] - a[i] <= k {
            j += 1;
        }
        sum += j - i - 1;
    }
    println!("{}", sum);
}
